pkg/segment/segexecution.go

```diff
@@ -156,10 +156,10 @@ func processQueryArithmeticNodeOp(queryOp *structs.QueryArithmetic, resMap map[u
 
 	*operationCounter++
 
-	return HelperQueryArithmeticAndLogical(queryOp, resMap)
+	return HelperQueryArithmeticAndLogical(queryOp, resMap, timeseriesApi)
 }
 
-func HelperQueryArithmeticAndLogical(queryOp *structs.QueryArithmetic, resMap map[uint64]*mresults.MetricsResult) (map[string]map[uint32]float64, error) {
+func HelperQueryArithmeticAndLogical(queryOp *structs.QueryArithmetic, resMap map[uint64]*mresults.MetricsResult, timeseriesApi bool) (map[string]map[uint32]float64, error) {
```

Comment:
nit: here and elsewhere, what do you think about renaming `timeseriesApi` to `labelsMustMatch` or similar?

Reply:
Hmm, yes I think I will refactor it to that. Seems generalized and makes more sense.

Reply:
I will rename it to `opLabelsMayNotMatch`, as for timeseries API, the operation labels does not need to match.

Reply:
I think that's still going to be kind of confusing, because sometimes "may not" means "must not" and in other contexts "may not" means "isn't required"

Reply:
My context here is convey that the operation labels does not need to match to perform the operation.